dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +354,228 @@ def fit(
         scorer=scorer,
         random_state=random_state,
     )
+
+
+# ----------------------------------------------------------------------------
+# Base class for scikit-learn compatible estimators using fit
+# ----------------------------------------------------------------------------
+
+
+class BaseIncrementalSearchCV(DaskBaseSearchCV):
+    """Base class for estimators using the incremental `fit`.
+
+    Subclasses must implement the following abstract methods
+
+    * _get_ids_and_args : Build the dict of
+       ``{model_id : (param_list, additional_calls)}``
+        that are eventually passed to `fit`
+    * _get_cv_results : Build the dict containing the CV results.
```

Comment:
I would expect this to be shared among implementations.  I think that the `history` output of the `fit` function should have all of the information here.  Are there choices that are likely to change between implementations?

Reply:
Yes, I think so. Initially, the structure of my `IncrementalRandomizedSearchCV` differed from `Hyperband`, as my results were simply `results = fit(...)`, and not `results = {model_id: fit(...) for ...}`.

Now that both have the consistent `results` type of `{model_id: (info, models, history)}`, more of this should be sharable.

same applies to https://github.com/dask/dask-ml/pull/356#discussion_r215591027

Reply:
In my implementation of `_get_cv_results`, the `bracket` key (`model_id` here I think) was only required to make it so the `model_id`s were unique across brackets. This was important for downstream processing; I wanted unique model IDs for each bracket, not shared (and isn't the cleanest code).

This solution can be used with the "consistent `results` type of `{model_id: (info, models, history)}`". The result I came up at [_hyperband.py#L315-L318][1] is something like 

``` python
for loop_id, (info, models, hist) in results.items():   # results defined in comment above
    for h in hist:
        h["model_id"] = "loop{}-{}".format(loop_id, h["model_id"])
```

[1]:https://github.com/dask/dask-ml/blob/4e142d75bd52dd351c90a59e488ad9a62e38daa3/dask_ml/model_selection/_hyperband.py#L315-L318